File_Adapter/CRUD/Push.cs

```diff
@@ -39,7 +39,18 @@ public override List<IObject> Push(IEnumerable<IObject> objects, string tag = ""
         {
             CreateFileAndFolder();
 
-            return base.Push(objects, tag, config);
+            List<IObject> objectsToPush = Config.CloneBeforePush ? objects.Select(x => x is BHoMObject ? ((BHoMObject)x).GetShallowClone() : x).ToList() : objects.ToList(); //ToList() necessary for the return collection to function properly for cloned objects
+
+            IEnumerable<IBHoMObject> bhomObjects = objectsToPush.Where(x => x is IBHoMObject).Cast<IBHoMObject>();
+
+            if (objects.Count() != objects.Count())
```

Comment:
`if (false)` ? A bit of a weird test 😋 
Also, shouldn't it be a warning instead of an error? From what I can see, this will execute correctly, just not with all the objects the user was expecting.